gremlin-integrations/templates/deployment.yaml

```diff
@@ -0,0 +1,117 @@
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: GREMLIN_TEAM_ID
+          {{- /* If we aren't managing this secret and a teamID was supplied, assume teamID is not in the external secret */}}
+          {{- if (and (not .Values.gremlin.secret.managed) (default .Values.gremlin.teamID .Values.gremlin.secret.teamID)) }}
+              value: {{ default .Values.gremlin.teamID .Values.gremlin.secret.teamID | quote }}
+          {{- else }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "gremlin.secretName" . }}
+                  key: GREMLIN_TEAM_ID
+          {{- end }}
+            - name: GREMLIN_SERVICE_URL
+              value: {{ include "gremlinServiceUrl" .}}
+          {{- if (eq (include "gremlin.secretType" .) "secret") }}
+            - name: GREMLIN_TEAM_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "gremlin.secretName" . }}
+                  key: GREMLIN_TEAM_SECRET
+          {{- else }}
+            - name: GREMLIN_TEAM_CERTIFICATE_OR_FILE
+              value: file:///var/lib/gremlin/cert/gremlin.cert
+            - name: GREMLIN_TEAM_PRIVATE_KEY_OR_FILE
+              value: file:///var/lib/gremlin/cert/gremlin.key
+          {{- end }}
+          {{- if .Values.gremlin.allowList }}
+            - name: GREMLIN_INTEGRATION_AGENT_ALLOW_LIST
+              value: {{ .Values.gremlin.allowList }}
+          {{- end }}
+          {{- if .Values.gremlin.proxy.url }}
+            - name: https_proxy
+              value: {{ .Values.gremlin.proxy.url }}
+            - name: no_proxy
+              value: $(KUBERNETES_SERVICE_HOST):$(KUBERNETES_SERVICE_PORT)
+          {{- end }}
+          {{- if .Values.ssl.certFile }}
+            - name: SSL_CERT_FILE
+              value: /etc/gremlin/ssl/certfile.pem
+          {{- end }}
+          {{- if .Values.ssl.certDir }}
+            - name: SSL_CERT_DIR
+              value: {{ .Values.ssl.certDir }}
+          {{- end }}
+          volumeMounts:
+        {{- if (eq (include "gremlin.secretType" .) "certificate") }}
+            - name: gremlin-cert
+              mountPath: /var/lib/gremlin/cert
+              readOnly: true
+        {{- end }}
+        {{- if .Values.ssl.certFile }}
+            - name: integrations-ssl-cert-file
+              mountPath: /etc/gremlin/ssl
+              readOnly: true
+        {{- end }}
+            - name: gremlin-logs
+              mountPath: /var/log/gremlin
+              readOnly: false
+      volumes:
+        - name: gremlin-logs
```

Comment:
NOTE; we ran into some problems with `/var/log/gremlin` being a hostPath on GKE under their COS Containerd.  Might be worth a test (see: https://github.com/gremlin/helm/pull/62 for reference)

Reply:
Unlike the gremlin agent, `gremlin-integrations` does not share this log information with sidecar processes, and so it does not need this hostPath mount.

Reply:
You are right. The logs are written and persisted since they are written to the standard output. This means that there is no need in mounting the logs since they aren't shared with other sidecar processes.